src/lib_ccx/ccx_decoders_structs.h

```diff
@@ -85,8 +85,8 @@ struct eia608_screen // A CC buffer
 	/** format of data inside this structure */
 	enum ccx_eia608_format format;
 	unsigned char characters[15][33];
-	enum ccx_decoder_608_color_code colors[15][33];
-	enum font_bits fonts[15][33]; // Extra char at the end for a 0
+	unsigned char colors[15][33];
```

Comment:
The only way I can think this fixing anything is by some other place using unsigned char instead of that enum. 
Let's correct the right place instead.

Reply:
It seems that it has to do with stuff being copied like so:
```c
memset(data->colors[i], context->settings->default_color, CCX_DECODER_608_SCREEN_WIDTH + 1);
```

Reply:
An enum is larger than 1 byte so it doesn't work

Reply:
An enum can be any size actually (and not all enums have to be the same size) - so let's use sizeof() where needed instead of removing the one hint we have about the type .-)

But good catch! Let's do _the_ best fix rather than _a_ fix :-)

Reply:
> An enum can be any size actually

how do you do that?

Reply:
Like this? https://stackoverflow.com/questions/13789462/is-possible-to-set-default-type-for-enum-to-be-unsigned-char

Reply:
Or when we memcpy, we will have to skip the most significant bytes of the enum?

Reply:
my bad as suggested by my previous comment, I always saw those `memset` as `memcpy` so that confused me.

Reply:
I know how to fix it

Reply:
Just use sizeof() :-) Don't replace it with #define or anything like that. Please please don't do "clever" stuff like copying only half the bits or anything like it :-)